sysutils/john/Portfile

```diff
@@ -1,115 +1,259 @@
 
-    checksums           rmd160  ea75c464d211691f00db7aec66e0503c1177866b \
-                        sha256  f5d123f82983c53d8cc598e174394b074be7a77756f5fb5ed8515918c81e7f3b \
-                        size    33638768
+    # NOTE: not all scripts were migrated to python3
+    # See:
+    #  - https://github.com/openwall/john/issues/4883
+    #  - https://github.com/openwall/john/pull/5309
+    set python2_branch          2.7
+    set python3_branch          3.11
 
-    use_configure       yes
-    configure.dir       ${worksrcpath}/src
+    set python2_version         [string map {. {}} ${python2_branch}]
+    set python3_version         [string map {. {}} ${python3_branch}]
 
-    # error We need minimum OpenCL 1.2 to build with OpenCL support. 
+    # NOTE: It has missed dependencies for py27-scapy / py311-scapy
+    # and py311-pysap which depends on py311-scapy
+    # See: https://trac.macports.org/ticket/68127
+    depends_run-append          port:p${perl_version}-compress-raw-lzma \
+                                port:p${perl_version}-digest-md4 \
+                                port:p${perl_version}-net-radius \
+                                port:p${perl_version}-netpacket \
+                                port:p${perl_version}-data-plist \
+                                port:perl${perl_version} \
+                                port:py${python2_version}-asn1 \
+                                port:py${python2_version}-plyvel \
+                                port:py${python3_version}-asn1crypto \
+                                port:py${python3_version}-bsddb3 \
+                                port:py${python3_version}-ldap3 \
+                                port:py${python3_version}-parsimonious \
+                                port:py${python3_version}-protobuf3 \
+                                port:python${python2_version} \
+                                port:python${python3_version} \
+                                port:ruby${ruby_version}
+
+    if {${subport} eq "john-jumbo"} {
+        depends_run-append      port:py${python2_version}-lxml \
+                                port:py${python2_version}-pycryptodome \
+    }
+
+    if {${subport} eq "john-jumbo-devel"} {
+        depends_run-append      port:py${python3_version}-asn1 \
+                                port:py${python3_version}-cbor2 \
+                                port:py${python3_version}-lxml \
+                                port:py${python3_version}-pycryptodome
+    }
+
+    configure.dir               ${worksrcpath}/src
+
+    # error We need minimum OpenCL 1.2 to build with OpenCL support.
     if {${os.platform} eq "darwin" && ${os.major} <= 11} {
-        configure.args-append      --disable-opencl
+        configure.args-append   --disable-opencl
+    }
+
+    build.dir                   ${worksrcpath}/src
+
+    variant native description {Enable CPU-specific optimizations} {
+        configure.args-delete   --enable-native-march
```

Comment:
shouldn't this be "append"?

Reply:
@reneeotten thanks, it should.

Thanks to catch it